### PR TITLE
 [IMP] product: clean forms and improve archiving

### PR DESCRIPTION
**Pull request by gla-odoo**
Several last cleaning in the product and product variant form :
- various labelling and design improvements
- sales, purchased and manufactured stat button (number of sales was
incorrect and click will now open normal sales analysis)
- now archiving related product.template if there is only one active product.product

Task-1880039  closes #28633

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/product.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from datetime import timedelta
+from datetime import datetime, timedelta, date
```
Comment (dbeguin) [inline addons/mrp/models/product.py]: Why is this needed ? This is not used in this file

File: addons/sale/models/product_template.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from datetime import datetime, timedelta, date
```
Comment (dbeguin) [inline addons/sale/models/product_template.py]: Same as previous comment

File: addons/stock_account/views/product_views.xml [hunk 3]
```diff
@@ -38,15 +38,17 @@
...
-                        <field name="standard_price"
-                            attrs="{'readonly':[('valuation','=','real_time')]}"
-                            widget='monetary' options="{'currency_field': 'currency_id', 'field_digits': True}"
-                            class="float-left"/>
-                        <div name="update_cost_price" class="col-lg-2">
-                            <button string="Update Cost" type="action"
-                            name="%(action_view_change_standard_price)d"
-                            attrs="{'invisible':['|', '|', ('valuation','!=', 'real_time'), ('cost_method', 'not in', ['average', 'standard']), ('product_variant_count', '>', 1)]}"
-                            class="oe_link oe_read_only"/>
+                        <div class="row  ml-0">
```
Comment (jem-odoo) [inline addons/stock_account/views/product_views.xml]: why this change ? To align the button with the field ? Isn't it possible with the 'oe_inline' class ?

**Comment by gla-odoo** [comment]
Thanks for your reviews. I pushed the changes...Not sure for the toggle_active overriding on product.py though...

**Comment by KangOl** [comment]
@robodoo r+

**Comment by KangOl** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
@robodoo retry

**Comment by tde-banana-odoo** [comment]
@robodoo r-

**Comment by tde-banana-odoo** [comment]
@robodoo r+
